docs/arcadia-users-group/20240305-github-templates/lesson.md

```diff
@@ -0,0 +1,71 @@
+On the New Repository launch screen, select the template you wish to start from in the "Repository template" drop down menu.
+In either case, fill out the rest of the new repository information and click the "Create repository" button.
+This will generate a new repository that already has all of the files and folders that were in the template.
+
+After your new repository is created, *create a new branch* before you start making any changes.
+Do your work in branches and when you are ready to merge your changes into the `main` branch, open a Pull Request and request a review.
+
+## Anatomy of a template: Python analysis template
+
+We can create a new repo from a template!
+But what is actually in the template and what do those files do?
+Below we go through each file in the [`python-analysis-template`](https://github.com/Arcadia-Science/python-analysis-template) and discuss their functions and how template users can interact with them.
+
+### Documentation and re-use
+
+- [README.md](https://github.com/Arcadia-Science/python-analysis-template/blob/main/README.md): This README has two sections. The first is an outline for the documentation for the new repository. It includes boiler plate sections and language that we expect will be standard across most projects, as well as TODO statements where specific details need to be changed or filled in. The second section covers developer notes detailing what the files in the repository do and tips for how to use them. The second section should eventually be deleted but is included for ease of use.
+- [envs/dev.yml](https://github.com/Arcadia-Science/python-analysis-template/tree/main/envs/dev.yml): By default, we recommend analysis repositories use conda to manage software installations and record software versions. The `envs` directory is where these yaml files are stored. The `dev.yml` file installs python, jupyter, pip, and linting software.
+- [LICENSE](https://github.com/Arcadia-Science/python-analysis-template/tree/main/LICENSE): The LICENSE file specifies the terms under which the software can be freely used, modified, and shared. Technically, software must have a license for it to be re-usable by others. The default license for all of our GitHub repositories is MIT, however the license may need to change depending on the software the project uses and the code it includes. Be sure to check with legal about your license before making a GitHub repository public.
+- [.github/PULL_REQUEST_TEMPLATE.md](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.github/PULL_REQUEST_TEMPLATE.md): The .github folder controls some behavior of the GitHub repository itself. The PULL_REQUEST_TEMPLATE.md is a text document where users can provide the text that they want to appear by default each time a new PR is opened. Often, these files remind users of the list of things that need to be done before a PR is read to be reviewed.
```

Comment:
Would it be worthwhile to also walk through the experience of opening a PR/ issue as part of a practical demo to illustrate what the templates provide? We could show a repo that hasn't used a template/ doesn't have the configs set up for PRs and issues, and show how much nicer it is to have some templates.

I wonder if it also might be good for folks to go through a dummy PR to understand what branch protections are and why we should be using them for new projects, as well as where/ how to see that linting checks are passing.

Reply:
both are great ideas and will for sure do these while leading the lesson. Same question as above -- to what extent do you think I should update the lesson with these examples vs. just showing them? I think a three paned PR template file,  opening a new PR in a repo with a template, and opening a new PR in a repo with no template is a great viz i'll put together  -- any others I should include?

Reply:
Looks good to me! Thanks for doing this!